src/expression.d

```diff
@@ -1564,10 +1585,11 @@ extern (C++) bool functionParameters(Loc loc, Scope* sc, TypeFunction tf, Type t
             }
             else
             {
-                //                arg = arg->isLvalue() ? callCpCtor(sc, arg) : valueNoDtor(arg);
+                //arg = doCopyOrMove(sc, arg);
```

Comment:
As far as I can see from the git history, when Walter had added imrovements for throwing exceptions during function arguments evaluation, this line was commented out. So today, it's surely a dead code and IMO it won't be enabled again.

I've just removed the empty else branch.